services/src/main/java/org/keycloak/services/resources/admin/UserProfileResource.java

```diff
@@ -49,7 +49,7 @@ public UserProfileResource(KeycloakSession session, AdminPermissionEvaluator aut
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     public String getConfiguration() {
-        auth.realm().requireViewRealm();
```

Comment:
Would make more sense `requireQuery`?

Reply:
I'm not sure.  Would someone ever have rights to see the realm configuration without having rights to query users?

Also, we will need a similar endpoint for the account console.  For that one, the UI will need to get the profile config without any admin rights at all.

So to me it made sense to allow the least restrictive permissions.  But I'm happy to change it if you disagree.

Reply:
My bad, forget about this comment. I was considering the changes we are about to expose the metadata for a user.